File: leetcode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.c
typedef struct {
    int* data;
    int top;
    int capacity;
} MyStack;


MyStack* myStackCreate() {
    MyStack* stack = (MyStack*)malloc(sizeof(MyStack));
    stack->data=(int*)malloc(sizeof(int)*100);
    stack->top = -1;
    stack->capacity=100;
    return stack;
}

void myStackPush(MyStack* obj, int x) {
      if (obj->top == obj->capacity - 1) {
        obj->capacity *= 2;
        obj->data = (int*)realloc(obj->data, sizeof(int) * obj->capacity);
    }
    obj->data[++obj->top] = x;
}

int myStackPop(MyStack* obj) {
    return obj->data[obj->top--];
}

int myStackTop(MyStack* obj) {
      return obj->data[obj->top];
}

bool myStackEmpty(MyStack* obj) {
      return obj->top == -1;
}

void myStackFree(MyStack* obj) {
        free(obj->data);
    free(obj);
}

/**
 * Your MyStack struct will be instantiated and called as such:
 * MyStack* obj = myStackCreate();
 * myStackPush(obj, x);
 
 * int param_2 = myStackPop(obj);
 
 * int param_3 = myStackTop(obj);
 
 * bool param_4 = myStackEmpty(obj);
 
 * myStackFree(obj);
*/

File: leetcode/Easy/0225-implement-stack-using-queues/0225-implement-stack-using-queues.py
class MyStack:

    queue1 = []
    queue2 = []
    
    def __init__(self):
        self.queue1 = []
        self.queue2 = []

    def push(self, x: int) -> None:
        self.queue1.append(x)

    def pop(self) -> int:
        x = 0
        while self.queue1:
            self.queue2.append(self.queue1.pop())
            x+=1
        for i in range(x-1):
            self.queue1.append(self.queue2.pop())
        return self.queue2.pop()

    def top(self) -> int:
        x = 0
        while self.queue1:
            self.queue2.append(self.queue1.pop())
            x+=1
        for i in range(x-1):
            self.queue1.append(self.queue2.pop())
        answer = self.queue2.pop()
        self.queue1.append(answer)
        return answer

    def empty(self) -> bool:
        return len(self.queue1)==0
        


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()